### PR TITLE
HOT FIX: only active users, fixed resource leak (refactoring as in download)

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes EASY-

#### When applied it will
* Fixes an LDAP resource leak
* Return 400-not-authenticated if a user has another dansState than ACTIVE
* 

#### How should this be manually tested?

* See previous PR's
* Try a user with `dansState==BLOCKED` (update default user with apache directory studio)

#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.solr4files/ApplicationWiring.scala [hunk 1]
```diff
@@ -37,21 +35,12 @@ class ApplicationWiring(configuration: Configuration)
   extends DebugEnhancedLogging
     with Vault
     with Solr
-    with LdapAuthenticationComponent {
+    with AuthenticationComponent {
```
Comment (PaulBoon) [inline src/main/scala/nl.knaw.dans.easy.solr4files/ApplicationWiring.scala:38]: It look s LDAP specific, so why removing Ldap from the name?

Comment (jo-pol) [reply src/main/scala/nl.knaw.dans.easy.solr4files/ApplicationWiring.scala:38]: I melted an abstract and ldap specific variant together as there is no use for another implementation. Tried it first with a single class in download and now applied the same here. Whenever we would replace ldap with something else, we wouldn't have to rename again.

File: src/main/scala/nl.knaw.dans.easy.solr4files/components/AuthenticationComponent.scala [hunk 2]
```diff
@@ -35,6 +46,65 @@ trait AuthenticationComponent extends DebugEnhancedLogging {
...
+        _ <- userIsActive(userAttributes)
+        roles = userAttributes.getOrElse("easyRoles", Seq.empty)
+      } yield User(userName,
+        isArchivist = roles.contains("ARCHIVIST"),
+        isAdmin = roles.contains("ADMIN"),
+        groups = userAttributes.getOrElse("easyGroups", Seq.empty)
+      )
+    }.recoverWith {
+      case t: InvalidUserPasswordException => Failure(t)
+      case t => Failure(AuthorisationNotAvailableException(t))
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.solr4files/components/AuthenticationComponent.scala]: Authorisation should be renamed into Authenticaion, 
but I'm already having merge conflicts with cleanup in #16

File: src/main/scala/nl.knaw.dans.easy.solr4files/ApplicationWiring.scala (hunk 1 above)
Comment (janvanmansum) [reply src/main/scala/nl.knaw.dans.easy.solr4files/ApplicationWiring.scala:38]: Hm, we would still have plenty of time to rename it then. It is not like we are planning to do that anytime soon, so @jo-pol 's argument doesn't really convince me. However, the more generic name doesn't bother me much, so let's just leave it in.

File: src/test/resources/debug-config/logback-service.xml [hunk 3]
```diff
@@ -2,24 +2,18 @@
...
-            <pattern>[%date{ISO8601}] %-5level %msg %n</pattern>
+            <pattern>[%date{ISO8601}] %-5level %msg%n</pattern>
         </encoder>
     </appender>
-
-    <appender name="FILE"
-              class="ch.qos.logback.core.rolling.RollingFileAppender">
+    <appender name="FILE" class="ch.qos.logback.core.FileAppender">
         <file>data/easy-solr4files-index.log</file>
-        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
```
Comment (rvanheest) [inline src/test/resources/debug-config/logback-service.xml:12]: why remove this policy? It's there in all modules, AFAIK!

File: src/test/resources/debug-config/logback.xml [hunk 4]
```diff
@@ -1,27 +1,18 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="UTF-8"?>
 <configuration>
 	<appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
-		<!-- encoders are assigned the type
-         ch.qos.logback.classic.encoder.PatternLayoutEncoder by default -->
 		<encoder>
-			<pattern>[%date{ISO8601}] %-5level %msg %n</pattern>
```
Comment (rvanheest) [inline src/test/resources/debug-config/logback.xml:7]: same here. and why is there a need to change the log pattern? these log files are used in the manual test phase and are (IMHO) fairly important. Also the changes below of turning off the logging at all seems completely weird. Either undo this change or give a very good explanation for doing this!

File: src/test/scala/nl.knaw.dans.easy.solr4files/SearchServletSpec.scala [hunk 5]
```diff
@@ -159,8 +159,8 @@ class SearchServletSpec extends TestSupportFixture
 
   it should "report no authentication available" in {
     get(s"/?text=nothing", headers = Map("Authorization" -> ("Basic " + Base64.encodeString("somebody:secret")))) {
-      status shouldBe SC_SERVICE_UNAVAILABLE
       body shouldBe "Authentication service not available, try anonymous search"
+      status shouldBe SC_SERVICE_UNAVAILABLE
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.solr4files/SearchServletSpec.scala:163]: what's the difference here? Just the order of the tests? Why???
